src/codeMarker.ts

```diff
@@ -2074,6 +2157,53 @@ export class CodeMarker implements vscode.TreeDataProvider<TreeEntry> {
         this.decorateWithUri(uri);
         this.refresh(uri);
     }
+
+    private cleanPartialAudits(uriToRemove: vscode.Uri) {
+        const relative = path.relative(this.workspacePath, uriToRemove.fsPath);
+        const entries = this.partialAuditedFiles.filter((file) => file.path === relative);
+        for (const entry of entries) {
+            const index = this.partialAuditedFiles.indexOf(entry);
+            if (index > -1) {
+                this.partialAuditedFiles.splice(index, 1);
+            }
+        }
+    }
+
+    private mergePartialAudits(): void {
+        const cleanedEntries: PartialAuditedFile[] = [];
+        // sort first by path and startLine for the merge to work
+        const sortedEntries = this.partialAuditedFiles.sort((a, b) => a.path.localeCompare(b.path) || a.location.startLine - b.location.startLine);
+        for (const entry of sortedEntries) {
+            // check if the current location is already partially audited
+            const partIdx = cleanedEntries.findIndex(
+                (file) =>
+                    // only merge entries for the same file
+                    file.path === entry.path &&
+                    // checks if the start is within bounds but the end is not
+                    ((file.location.startLine <= entry.location.startLine && file.location.endLine >= entry.location.startLine) ||
+                        // checks if the end is within bounds but the start is not
+                        (file.location.startLine <= entry.location.endLine && file.location.endLine >= entry.location.endLine) ||
+                        // checks if the location includes the entry
+                        (file.location.startLine >= entry.location.startLine && file.location.endLine <= entry.location.endLine)),
+            );
+            // update entry if necessary
+            if (partIdx > -1) {
+                const foundLocation = cleanedEntries[partIdx].location;
+                if (foundLocation.endLine < entry.location.endLine) {
+                    foundLocation.endLine = entry.location.endLine;
+                }
+                if (foundLocation.startLine > entry.location.startLine) {
+                    foundLocation.startLine = entry.location.startLine;
+                }
+
+                cleanedEntries[partIdx].location = foundLocation;
+            } else {
+                cleanedEntries.push(entry);
+            }
+        }
+
+        this.partialAuditedFiles = cleanedEntries;
+    }
```

Comment:
I would say that we can remove this as it doesn't cover all cases where we could merge, that it adds a bit too much complexity and the gain is that we would save a few entries. I don't expect the number of partially audited file entries to be very large as to require something like this for now

Reply:
What are the cases it doesn't cover? Just curiosity what I have missed.

I added this merging logic because the rendering looks awful and not because of the amount of entries. If two entries overlap the green background overlaps and then starts to look thicker.
![image](https://github.com/trailofbits/vscode-weaudit/assets/4057473/0861b803-a431-43f4-ba1d-53836a5f3faf)


Depending on what are a lot of entries for you. Currently, I am auditing a code base with over 1k lines in a single file so with the merging I am currently at ~65 entries. I don't think this is a lot.

Reply:
I now see why you wrote this!
The case I noticed that wasn't covered is if you have contiguous but non-intersecting regions say lines 1-5 and 6-8, I assumed this would also want to merge the two regions.

Reply:
Yep this is on purpose because I just wanted to fix the UI bug. 
But I'll add that adjacent regions also merge into a bigger one with the location change soon

Reply:
it merges now adjacent entries too